babel/src/main/java/org/apache/calcite/sql/babel/SqlExecMacroParam.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql.babel;
+
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlIdentifier;
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlOperator;
+import org.apache.calcite.sql.SqlSpecialOperator;
+import org.apache.calcite.sql.SqlWriter;
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.util.ImmutableNullableList;
+
+import java.util.List;
+
+/**
+ * Parse tree for {@code SqlExecMacroParam} SqlCall.
+ */
+public class SqlExecMacroParam extends SqlCall {
+  private static final SqlSpecialOperator OPERATOR =
+      new SqlSpecialOperator("PARAM EQUAL", SqlKind.OTHER);
+
+  private final SqlIdentifier name;
+  private final SqlNode value;
+
+  /**
+   * Create an {@code SqlExecMacroParam} without name.
+   *
+   * @param pos  Parser position, must not be null
+   * @param value  Value of the parameter
+   */
+  public SqlExecMacroParam(SqlParserPos pos, SqlNode value) {
+    this(pos, /*name=*/null, value);
+  }
+
+  /**
+   * Create an {@code SqlExecMacroParam}.
+   *
+   * @param pos  Parser position, must not be null
+   * @param name  Name of the parameter
+   * @param value  Value of the parameter
+   */
+  public SqlExecMacroParam(SqlParserPos pos, SqlIdentifier name, SqlNode value) {
+    super(pos);
+    this.name = name;
+    this.value = value;
+  }
+
+  @Override public SqlOperator getOperator() {
+    return OPERATOR;
+  }
+
+  @Override public List<SqlNode> getOperandList() {
+    return ImmutableNullableList.of(name, value);
+  }
+
+  @Override public void unparse(final SqlWriter writer, final int leftPrec,
+      final int rightPrec) {
+    if (this.name != null) {
+      name.unparse(writer, 0, 0);
+      writer.keyword("=");
```

Comment:
I believe this should be unparsed as a separator:
writer.sep("=");

Reply:
It's good that you brought this to my attention. I have done some investigation into whether "=" is a separator or a keyword. Judging by [the Java doc of sep(String)](https://github.com/googleinterns/calcite/blob/master/core/src/main/java/org/apache/calcite/sql/SqlWriter.java#L493), it seems like sep(String) are used for list separator. I think "=" would be more of a keyword than a separator in this standard. Would you agree with my judgement or is there something I missed?

Reply:
I have mostly seen "=" unparsed as a separator in existing code. That said, if it is ambiguous and you feel this is more appropriate, that's fine too.

Reply:
I would appreciate it, if you can point me to a few examples on where past contributor use "=" as separator.